src/services/admin.rs

```diff
@@ -459,17 +464,29 @@ impl AdminService {
                 ),
             };
             let mut database = self.database.lock().await;
-            node.id = database.create_node(node.clone())?;
+            dbg!("Locked the database!");
+            node.id = database.create_node(node.clone())
+                .map_err(|err| {
+                    error!("Unable to create node in Sensei Database");
+                    database.clear_pending_node();
+                    err
+                })?;
+            
+            
+            dbg!("Created?");
             node
         };
+        dbg!("Node is created in the database");
 
-        let lightning_node = self.get_node(node.clone(), passphrase).await?;
+        let lightning_node = self.get_node(node.clone(), passphrase).await.unwrap();
```

Comment:
I have the explicit unwrap here because it felt like a panic at this stage, if it failed, is pretty acceptable.  Not sure - the previous behavior was, "Fail silently" and this seems _better_ but not necessarily optimal.